openpype/modules/kitsu/utils/sync_service.py

```diff
@@ -335,14 +343,25 @@ def _new_task(self, data):
         """Create new task into OP DB."""
         # Get project entity
         set_op_project(self.dbcon, data["project_id"])
+        project_name = self.dbcon.active_project()
 
         # Get gazu entity
         task = gazu.task.get_task(data["task_id"])
 
         # Find asset doc
-        asset_doc = self.dbcon.find_one(
-            {"type": "asset", "data.zou.id": task["entity"]["id"]}
+        parent_name = task["entity"]["name"]
+        parent_zou_id = task["entity"]["id"]
+        asset_docs = get_assets(
+            project_name,
+            asset_names=[parent_name],
+            fields=["_id", "data.zou.id", "data.tasks"]
         )
+        asset_doc = None
+        for _asset_doc in asset_docs:
+            doc_zou_id = _asset_doc.get("data", {}).get("zou", {}).get("id")
+            if doc_zou_id == parent_zou_id:
+                asset_doc = _asset_doc
+                break
```

Comment:
Don't you have an API function `get_single_asset` like function? It'd be better than having to filter for every module no?

Or use your `get_asset_by_name`? Doing `get_asset_by_name(task['entity']['name'])`

Reply:
I'm ok with that. I wasn't sure if the check of `zou.id` should stay?

Reply:
It is not supposed to change, and the name is supposed to be unique in both DB (zou and OP), then we can drop this UUID check.

Reply:
Done